### PR TITLE
feat(store): #13 - Basic ngrx store setup

### PR DESCRIPTION
**Pull request by naveenvignesh5**
## Steps To Implement Feature

1. Add ngRx module to application `npm install @ngrx/store --save`
2. Setup basic store with dummy board reducer, board action with optional board selector
3. Added the Store to `app.module.ts` module declaration.

### CODE REVIEW

File: web/src/models/board.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export class Board {
```
Comment (ramsunvtech) [inline web/src/models/board.ts]: keep it as BoardItem

File: web/src/models/board.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+  id: number;
+  title: string;
+  boardId: number;
+}
+
+export class BoardState {
```
Comment (ramsunvtech) [inline web/src/models/board.ts]: BoardListState

File: web/src/redux/Store.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import { StoreModule } from '@ngrx/store';
```
Comment (ramsunvtech) [inline web/src/redux/Store.ts]: keep it as store/index.ts

**Comment by ramsunvtech** [comment]
@naveenvignesh5, Please update the commit message like below and you should have multiple commits.

```
* **store** - Store Changes
* **action/<actionName>** - Redux Action Changes
* **selector/<selectorName>** - Redux Store Selector Changes
* **reducer/<reducerName>** - Redux Store Reducer Changes
```

or one short commit message like below.

```
feat(store): #ID - message
```

File: web/src/models/board.ts (hunk 2 above)
Comment (naveenvignesh5) [reply web/src/models/board.ts]: Changed name

File: web/src/redux/Store.ts (hunk 3 above)
Comment (naveenvignesh5) [reply web/src/redux/Store.ts]: moved to store folder and put it `index.ts`

**Comment by naveenvignesh5** [comment]
> @naveenvignesh5, Please update the commit message like below and you should have multiple commits.
> 
> ```
> * **store** - Store Changes
> * **action/<actionName>** - Redux Action Changes
> * **selector/<selectorName>** - Redux Store Selector Changes
> * **reducer/<reducerName>** - Redux Store Reducer Changes
> ```
> 
> or one short commit message like below.
> 
> ```
> feat(store): #ID - message
> ```

Updated the comment message @ramsunvtech

File: web/src/models/board.ts (hunk 1 above)
Comment (naveenvignesh5) [reply web/src/models/board.ts]: Updated to `BoardItem`

File: web/src/app/app.module.ts [hunk 4]
```diff
@@ -3,14 +3,12 @@ import { NgModule } from '@angular/core';
 
 import { AppComponent } from './app.component';
 
+import { Store } from '../redux/store';
```
Comment (ramsunvtech) [inline web/src/app/app.module.ts]: keep it in `src/store`

File: web/src/redux/actions/board.action.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { createAction, props } from '@ngrx/store';
```
Comment (ramsunvtech) [inline web/src/redux/actions/board.action.ts]: redux directory can be removed and keep it in store.

Comment (naveenvignesh5) [reply web/src/redux/actions/board.action.ts]: fixed

File: web/src/app/app.module.ts (hunk 4 above)
Comment (naveenvignesh5) [reply web/src/app/app.module.ts]: fixed

**Review by ramsunvtech — APPROVED** [review]

**Pull request merged**